citations/citations-tool/tool/src/java/org/sakaiproject/citation/tool/CitationHelperAction.java

```diff
@@ -2297,277 +2297,279 @@ private String buildEditAttachmentPanelContext(VelocityPortlet portlet, Context
+				context.put("exportUrlAll", exportUrlAll);
+
+				Integer listPageSize = (Integer) state.getAttribute(STATE_LIST_PAGE_SIZE);
+				if (listPageSize == null) {
+					listPageSize = defaultListPageSize;
+					state.setAttribute(STATE_LIST_PAGE_SIZE, listPageSize);
+				}
+				context.put("listPageSize", listPageSize);
+
+				CitationIterator newIterator = citationCollection.iterator();
+				CitationIterator oldIterator = (CitationIterator) state.getAttribute(STATE_LIST_ITERATOR);
+				if (oldIterator == null) {
+					newIterator.setPageSize(listPageSize.intValue());
+					newIterator.setStart(0);
+				} else {
+					newIterator.setPageSize(listPageSize.intValue());
+					newIterator.setStart(oldIterator.getStart());
 //				newIterator.setPage(oldIterator.getPage());
-			}
-			context.put("citations", newIterator);
-			context.put("citationCollectionId", citationCollection.getId());
-			if(! citationCollection.isEmpty())
-			{
-				context.put("show_citations", Boolean.TRUE);
+				}
+				context.put("citations", newIterator);
+				context.put("citationCollectionId", citationCollection.getId());
+				if (!citationCollection.isEmpty()) {
+					context.put("show_citations", Boolean.TRUE);
 
 //				int page = newIterator.getPage();
 //				int pageSize = newIterator.getPageSize();
-				int totalSize = citationCollection.size();
+					int totalSize = citationCollection.size();
 
-				int start = newIterator.getStart();
-				int end = newIterator.getEnd();
+					int start = newIterator.getStart();
+					int end = newIterator.getEnd();
 //				int start = page * pageSize + 1;
 //				int end = Math.min((page + 1) * pageSize, totalSize);
 
-				Integer[] position = { new Integer(start+1) , new Integer(end), new Integer(totalSize)};
-				String showing = (String) rb.getFormattedMessage("showing.results", position);
-				context.put("showing", showing);
-			}
-			state.setAttribute(STATE_LIST_ITERATOR, newIterator);
-			
-			// constant schema identifier
-			context.put( "titleProperty", Schema.TITLE );
+					Integer[] position = {new Integer(start + 1), new Integer(end), new Integer(totalSize)};
+					String showing = (String) rb.getFormattedMessage("showing.results", position);
+					context.put("showing", showing);
+				}
+				state.setAttribute(STATE_LIST_ITERATOR, newIterator);
 
-			int requestStateId = preserveRequestState(state, new String[]{CitationHelper.RESOURCES_REQUEST_PREFIX, CitationHelper.CITATION_PREFIX});
-			context.put("requestStateId", requestStateId);
+				// constant schema identifier
+				context.put("titleProperty", Schema.TITLE);
 
-			
+				int requestStateId = preserveRequestState(state, new String[]{CitationHelper.RESOURCES_REQUEST_PREFIX, CitationHelper.CITATION_PREFIX});
+				context.put("requestStateId", requestStateId);
+
+
+			}
+
+			return TEMPLATE_NEW_RESOURCE;
+		}
+		catch (RuntimeException e){
+			throw new RuntimeException(e + "Citation Collection Id: " + citationCollectionId);
```

Comment:
I don't think you want this as it means you lose the original Exception object, shouldn't it be:

    throw new RuntimeException("Citation collection ID: "+ citationCollectionId, e);

that way you nest the exceptions, rather than calling `toString()` on the first one?